bloom/generators/common.py

```diff
@@ -134,6 +145,21 @@ def package_conditional_context(ros_distro):
             }
 
 
+def evaluate_package_conditions(package, ros_distro):
+    """
+    Evaluates a package's conditional fields if it supports them.
+
+    :param package: The package to be evaluated.
+    :param ros_distro: The codename of the rosdistro use for context.
+    :returns: None. The given package will be modified.
+    """
+    # Conditional fields were introduced in package format 3.
+    # Earlier formats should have their conditions evaluated with no context so
+    # the evaluated_condition is set to True in all cases.
+    if package.package_format >= 3:
+        package.evaluate_conditions(package_conditional_context(ros_distro))
```

Comment:
Accorint to the latest implement of [catkin_pkg](https://github.com/ros-infrastructure/catkin_pkg/blob/master/src/catkin_pkg/package.py#L183), the package.evaluate_conditions(package_conditional_context(ros_distro)) would work when package_format equal 2. As we can see from [there](https://github.com/ros-infrastructure/catkin_pkg/blob/master/src/catkin_pkg/package.py#L344), the package.evaluate_conditions change the method evaluated_condition of every dependence to dependence object's attribute. Consider we use this attribute lately, I think this function should be also executed when package_format equals 2.